lib/portage/gpkg.py

```diff
@@ -1523,10 +1566,30 @@ def _verify_binpkg(self, metadata_only=False):
                     f"Cannot read tar file: {self.gpkg_file}"
                 )
 
-            # Check gpkg header
-            if self.gpkg_version not in container_files:
+            # Check if gpkg version file exists in any place
+            if self.gpkg_version not in [os.path.basename(f) for f in container_files]:
```

Comment:
Use a generator rather than list comprehension here, as there should be no need to inline the result into a list first (i.e. change `[...]` to `(...)`).